Resources/Computing/SingularityComputingElement.py

```diff
@@ -342,14 +379,18 @@ def submitJob(self, executableFile, proxy=None, **kwargs):
     withCVMFS = os.path.isdir("/cvmfs")
     innerCmd = os.path.join(self.__innerdir, "dirac_container.sh")
     cmd = [self.__singularityBin, "exec"]
-    cmd.extend(["-c", "-i", "-p"])
-    cmd.extend(["-W", baseDir])
+    cmd.extend(["--contain"])  # use minimal /dev and empty other directories (e.g. /tmp and $HOME)
+    cmd.extend(["--ipc", "--pid"])  # run container in new IPC and PID namespaces
+    cmd.extend(["--workdir", baseDir])  # working directory to be used for /tmp, /var/tmp and $HOME
     if withCVMFS:
-      cmd.extend(["-B", "/cvmfs"])
+      cmd.extend(["--bind", "/cvmfs"])
     if 'ContainerBind' in self.ceParameters:
       bindPaths = self.ceParameters['ContainerBind'].split(',')
       for bindPath in bindPaths:
-        cmd.extend(["-B", bindPath.strip()])
+        if len(bindPath.split(':::')) == 1:
+          cmd.extend(["--bind", bindPath.split(':::')[0].strip()])
+        elif len(bindPath.split(':::')) == 2:
+          cmd.extend(["--bind", bindPath.split(':::')[0].strip(), bindPath.split(':::')[0].strip()])
```

Comment:
Did yo mean:

        if len(bindPath.split(':::')) == 1:
          cmd.extend(["--bind", bindPath.strip()])
        elif len(bindPath.split(':::')) == 2:
          cmd.extend(["--bind", bindPath.split(':::')[0].strip(), bindPath.split(':::')[1].strip()])

Reply:
Yes, that's better. Applied.